File: app/src/main/java/com/szeptun/shoppinglist/domain/GetListsByState.java
package com.szeptun.shoppinglist.domain;

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository;
import com.szeptun.shoppinglist.entity.ListState;
import com.szeptun.shoppinglist.entity.ShoppingList;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Flowable;

public class GetListsByState {

    @Inject
    GetListsByState() {
    }

    @Inject
    ShoppingListRepository repository;

    public Flowable<List<ShoppingList>> execute(ListState listState) {
        return repository.getListsByState(listState);
    }
}

File: app/src/main/java/com/szeptun/shoppinglist/domain/GetListsByState.kt
package com.szeptun.shoppinglist.domain

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository
import com.szeptun.shoppinglist.entity.ListState
import com.szeptun.shoppinglist.entity.ShoppingList
import io.reactivex.Flowable
import javax.inject.Inject

class GetListsByState @Inject constructor(private val repository: ShoppingListRepository) {

    fun execute(listState: ListState): Flowable<List<ShoppingList>> =
        repository.getListsByState(listState)
}